service/src/main/kotlin/fi/espoo/evaka/pedagogicaldocument/PedagogicalDocumentController.kt

```diff
@@ -0,0 +1,259 @@
+        Audit.PedagogicalDocumentUpdate.log(body.childId)
+        accessControl.requirePermissionFor(user, Action.Child.CREATE_PEDAGOGICAL_DOCUMENT, body.childId.raw)
+        val doc = createDocument(db, user, body)
+        return ResponseEntity.ok(doc)
+    }
+
+    @PutMapping("/{documentId}")
+    fun updatePedagogicalDocument(
+        db: Database.Connection,
+        user: AuthenticatedUser,
+        @PathVariable documentId: PedagogicalDocumentId,
+        @RequestBody body: PedagogicalDocumentPostBody
+    ): ResponseEntity<PedagogicalDocument> {
+        Audit.PedagogicalDocumentUpdate.log(documentId)
+        accessControl.requirePermissionFor(user, Action.Child.UPDATE_PEDAGOGICAL_DOCUMENT, body.childId.raw)
+        val doc = updateDocument(db, user, body, documentId)
+        return ResponseEntity.ok(doc)
+    }
+
+    @GetMapping("/{documentId}")
+    fun getPedagogicalDocument(
+        db: Database.Connection,
+        user: AuthenticatedUser,
+        @PathVariable documentId: PedagogicalDocumentId
+    ): ResponseEntity<PedagogicalDocument> {
+        val childId = findRelatedChild(db, documentId)
+        Audit.PedagogicalDocumentRead.log(childId)
+        accessControl.requirePermissionFor(user, Action.Child.READ_PEDAGOGICAL_DOCUMENT, childId.raw)
+        val doc = getPedagogicalDocument(db, documentId)
+        return ResponseEntity.ok(doc)
+    }
+
+    @GetMapping("/child/{childId}")
+    fun getChildPedagogicalDocuments(
+        db: Database.Connection,
+        user: AuthenticatedUser,
+        @PathVariable childId: ChildId
+    ): ResponseEntity<List<PedagogicalDocument>> {
+        Audit.PedagogicalDocumentUpdate.log(childId)
+        accessControl.requirePermissionFor(user, Action.Child.READ_PEDAGOGICAL_DOCUMENT, childId.raw)
+        val docs = findPedagogicalDocumentsByChild(db, childId)
+        return ResponseEntity.ok(docs)
+    }
+
+    @DeleteMapping("/{documentId}")
+    fun deletePedagogicalDocument(
+        db: Database.Connection,
+        user: AuthenticatedUser,
+        @PathVariable documentId: PedagogicalDocumentId
+    ): ResponseEntity<Unit> {
+        val childId = findRelatedChild(db, documentId)
+        Audit.PedagogicalDocumentUpdate.log(childId)
+        accessControl.requirePermissionFor(user, Action.Child.DELETE_PEDAGOGICAL_DOCUMENT, childId.raw)
+        deleteDocument(db, documentId)
+        return ResponseEntity.ok().build()
+    }
+}
+
+data class Attachment(
+    val id: AttachmentId,
+    val name: String,
+    val contentType: String
+)
+
+data class PedagogicalDocument(
+    val id: PedagogicalDocumentId,
+    val childId: ChildId,
+    val description: String,
+    val attachment: Attachment?,
+    val created: HelsinkiDateTime,
+    val updated: HelsinkiDateTime
+)
+
+data class PedagogicalDocumentPostBody(
+    val childId: ChildId,
+    val description: String,
+)
+
+fun findRelatedChild(db: Database.Connection, documentId: PedagogicalDocumentId): ChildId {
```

Comment:
All the queries in this file use a different pattern than we normally use.

In this file:

```
fun findStuff(db: Database.Connection)
```

- queries are not extension functions
- every query manages its own transactions

More common pattern:

```
fun Database.Read.findStuff()
```

- query is an extension function
- query *requires a transaction*, but doesn't manage one by itself